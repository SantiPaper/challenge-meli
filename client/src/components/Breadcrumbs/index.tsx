import { Categories } from "../../types/items";
import { StyledBreadcrumbs } from "./style";

type Props = {
  categories: Categories;
};
export const Breadcrumbs = ({ categories }: Props) => {
  return (
    <StyledBreadcrumbs>
      <ul className="breadcrumbs">
        {categories.map((catergory) => (
          <li key={catergory} className="breadcrumbs__item">
            <a href="#" className="breadcrumbs__item__link">
              {catergory}
            </a>
          </li>
        ))}
      </ul>
    </StyledBreadcrumbs>
  );
};
