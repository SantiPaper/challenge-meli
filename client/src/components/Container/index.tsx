import { FC, PropsWithChildren } from "react";
import { StyledContainer } from "./style";

export const Container: FC<PropsWithChildren> = ({ children }) => {
  return <StyledContainer>{children}</StyledContainer>;
};
