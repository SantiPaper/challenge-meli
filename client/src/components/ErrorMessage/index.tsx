import { StyledErrorMessage } from "./style";

type Props = {
  error: string | null;
};

export const ErrorMessage = ({ error }: Props) => {
  return (
    <StyledErrorMessage>
      <h2 className="text">{error}</h2>
    </StyledErrorMessage>
  );
};
