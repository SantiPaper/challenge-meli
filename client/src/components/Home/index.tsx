import { StyledHome } from "./style";

export const Home = () => {
  return (
    <StyledHome>
      <h1 className="hero">
        Aun no realizaste tu b√∫squeda, te alentamos a que lo hagas!
      </h1>
    </StyledHome>
  );
};
