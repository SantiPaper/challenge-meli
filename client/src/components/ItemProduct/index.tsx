import { StyledItemProduct } from "./style";
import imgShipping from "../../../public/images/shipping.png";

type Props = {
  title: string;
  price: number;
  img: string;
  shipping: boolean;
  id: string;
};
export const ItemProduct = ({ title, price, img, shipping, id }: Props) => {
  return (
    <StyledItemProduct id={id}>
      <img
        className="product__img"
        src={img}
        alt={title}
        width={180}
        height={180}
      />

      <div className="product__info">
        <div className="product__info__price-container">
          <p className="product__info__price-container__price">$ {price}</p>
          {shipping && <img src={imgShipping} alt="" width={15} height={15} />}
        </div>
        <h3 className="product__info__title">{title}</h3>
      </div>
    </StyledItemProduct>
  );
};
