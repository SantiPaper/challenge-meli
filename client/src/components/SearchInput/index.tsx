import { StyledSearchInput } from "./style";
import { useNavigate } from "react-router-dom";
import { ChangeEvent, useState } from "react";

export const SearchInput = () => {
  const navigate = useNavigate();
  const [search, setSearch] = useState<string>("");

  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
    setSearch(ev.target.value);
  };

  const handleSubmit = () => {
    navigate(`/items?search=${search}`);
  };

  return (
    <StyledSearchInput onSubmit={handleSubmit}>
      <label htmlFor="search" className="sr-only">
        Buscar productos
      </label>
      <input
        onChange={handleChange}
        className="search__input"
        placeholder="Nunca dejes de buscar"
        type="search"
        id="search"
        name="search"
      />
      <button className="search__button" aria-label="Buscar" type="submit">
        <img width={20} src="/images/search.png" alt="Buscar" />
      </button>
    </StyledSearchInput>
  );
};
