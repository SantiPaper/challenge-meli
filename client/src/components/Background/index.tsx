import { FC, PropsWithChildren } from "react";
import { Container } from "../Container";
import { StyledBackground } from "./style";

export const Background: FC<PropsWithChildren> = ({ children }) => {
  return (
    <Container>
      <StyledBackground>{children}</StyledBackground>
    </Container>
  );
};
