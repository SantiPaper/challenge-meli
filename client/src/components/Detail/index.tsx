import { useLocation } from "react-router-dom";
import { useFetch } from "../../hooks/useFetch";
import { useProductsContext } from "../../hooks/useProductsContext";
import { Breadcrumbs } from "../Breadcrumbs";
import { Loading } from "../Loading";
import { Background } from "../Background";
import { StyledDetail } from "./style";

export const Detail = () => {
  const location = useLocation();

  const { state, handleFormat } = useProductsContext();

  const product = state.product.item;

  const categories = state.product.categories;

  const { loading } = useFetch({
    status: "/",
    payload: location.pathname.split("/")[2],
  });

  return (
    <StyledDetail>
      {loading ? (
        <Loading />
      ) : (
        <>
          <Breadcrumbs categories={categories} />
          <Background>
            <div className="detail__info">
              <img
                className="detail__info__img"
                src={product.picture}
                alt={product.title}
                width={680}
                height={680}
              />

              <div className="detail__info__aside">
                <p className="detail__info__aside__condition">
                  {product.condition === "new" ? "Nuevo" : "Usado"}
                </p>
                <h1 className="detail__info__aside__title">{product.title}</h1>
                <p className="detail__info__aside__price">
                  {handleFormat(product.price.amount)}
                </p>
                <button className="detail__info__aside__button">Comprar</button>
              </div>
            </div>
            <div className="detail__description">
              <h3 className="detail__description__title">
                Descripci√≥n del producto
              </h3>
              <p className="detail__description__text">{product.description}</p>
            </div>
          </Background>
        </>
      )}
    </StyledDetail>
  );
};
