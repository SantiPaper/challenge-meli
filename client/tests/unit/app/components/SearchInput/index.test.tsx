import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { SearchInput } from "../../../../../src/components/SearchInput";
import { vi } from "vitest";

describe("Input de busqueda", () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <SearchInput />
      </BrowserRouter>
    );
  });

  test("Renderiza", () => {
    const { container } = render(
      <BrowserRouter>
        <SearchInput />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("Existe placeholder", () => {
    const input = screen.getByPlaceholderText("Nunca dejes de buscar");
    expect(input).toBeInTheDocument();
  });

  test("Existe el botón del componente", () => {
    const button = screen.getByRole("button", { name: "Buscar" });
    expect(button).toBeInTheDocument();
  });

  test("Permite escribir en el input de búsqueda", async () => {
    const inputElement = screen.getByPlaceholderText("Nunca dejes de buscar");

    await userEvent.type(inputElement, "Zapatos");

    expect(inputElement).toHaveValue("Zapatos");
  });
  test("Se puede enviar el query", async () => {
    const button = screen.getByRole("button");
    const form = screen.getByTestId("form");
    const fn = vi.fn();
    form.onsubmit = fn;
    await userEvent.click(button);
    expect(form.onsubmit).toHaveBeenCalledTimes(1);
  });
});
