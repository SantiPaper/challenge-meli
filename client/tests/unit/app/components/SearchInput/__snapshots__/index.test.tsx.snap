// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input de busqueda > Renderiza 1`] = `
<div>
  <form
    class="sc-beySPh bJRPyz"
    data-testid="form"
  >
    <label
      class="sr-only"
      for="search"
    >
      Buscar productos
    </label>
    <input
      class="search__input"
      id="search"
      name="search"
      placeholder="Nunca dejes de buscar"
      type="search"
    />
    <button
      aria-label="Buscar"
      class="search__button"
      type="submit"
    >
      <img
        alt="Buscar"
        src="/images/search.png"
        width="20"
      />
    </button>
  </form>
</div>
`;
